code/game/objects/items/RCD.dm

```diff
@@ -766,9 +771,19 @@ RLD
 	. = ..()
 	if(!range_check(A,user))
 		return
-	if(target_check(A,user))
-		user.Beam(A,icon_state="rped_upgrade", time = 3 SECONDS)
-	rcd_create(A,user)
+	pre_attack(A, user)
+
+/obj/item/construction/rcd/arcd/afterattack_secondary(atom/A, mob/user)
+	if(!range_check(A,user))
+		return
+	pre_attack_secondary(A, user)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+
+/obj/item/construction/rcd/arcd/rcd_create(atom/A, mob/user)
+	. = ..()
+	user.Beam(A,icon_state="rped_upgrade", time = 3 SECONDS)
+	return
```

Comment:
Remove this, trailing returns are pointless.

Reply:
I'll have it return what the original proc returned, which is probably what I intended it to do.